### DISCUSSION TITLE
in PrimeNG table scrollable=true sticky header is not working

### DISCUSSION THREAD

**Discussion by Rajput94** [primefaces/community#133] [PrimeNG] [answered]
<p-table
  [value]="customers"
  [scrollable]="true"
  [rows]="20"
  [paginator]="true"
  [showCurrentPageReport]="true"
>
  <ng-template pTemplate="header">
    <tr>
      <th>Name</th>
      <th>Country</th>
      <th>Representative</th>
      <th>Status</th>
    </tr>
  </ng-template>
  <ng-template pTemplate="body" let-customer>
    <tr>
      <td>
        {{ customer.name }}
      </td>
      <td>
        <span class="image-text">{{ customer.country.name }}</span>
      </td>
      <td>
        <span class="image-text">{{ customer.representative.name }}</span>
      </td>
      <td>
        <span [class]="'customer-badge status-' + customer.status">{{
          customer.status
        }}</span>
      </td>
    </tr>
  </ng-template>
</p-table>

  styles: [`
        :host ::ng-deep .p-datatable .p-datatable-thead > tr > th {
            position: -webkit-sticky;
            position: sticky;
            top: 0px;
        }

        @media screen and (max-width: 64em) {
            :host ::ng-deep .p-datatable .p-datatable-thead > tr > th {
                top: 0px;
            }
        }
  `]

**Comment by cetincakiroglu** [comment]
Hi,

Could you please create a Stackblitz example so we can identify the issue?

**Reply by Rajput94** [reply]
https://stackblitz.com/edit/primeng-tablesticky-demo-vcidwy?file=src%2Fapp%2Fapp.component.html

**Reply by Rajput94** [reply]
Is this URL ok ?

**Reply by cetincakiroglu** [reply, marked as answer]
Hi,

The scrollable feature breaks the sticky, it's true because it does not have support for the sticky. Sticky is not an actual feature of primeng is a little CSS tweak actually. As an alternative, you can achieve the same thing by `vertical scroll` https://primeng.org/table#vertical-scroll or you can try to make sticky work by custom CSS.

Here is an implementation of vertical scroll to your example https://stackblitz.com/edit/primeng-tablesticky-demo-vcidwy?file=src%2Fapp%2Fapp.component.html
